cpp/modmesh/view/R3DWidget.hpp

```diff
@@ -46,24 +46,22 @@
 namespace modmesh
 {
 
-class RScene
-    : public Qt3DCore::QEntity
+class RScene : public Qt3DCore::QEntity
 {
-
 public:
 
-    explicit RScene(Qt3DCore::QNode * parent = nullptr)
-        : Qt3DCore::QEntity(parent)
-        , m_controller(new ROrbitCameraController(this))
+    explicit RScene(QNode * parent = nullptr)
+        : QEntity(parent)
     {
+        m_controller = new ROrbitCameraController(this);
     }
 
-    Qt3DExtras::QAbstractCameraController const * controller() const { return m_controller; }
-    Qt3DExtras::QAbstractCameraController * controller() { return m_controller; }
+    CameraController * controller() const { return m_controller; }
+    Qt3DExtras::QAbstractCameraController * qtController() const { return m_controller->asQtCameraController(); }
 
-    void setCameraController(Qt3DExtras::QAbstractCameraController * controller)
+    void setCameraController(CameraController * controller)
     {
-        m_controller->deleteLater();
+        qtController()->deleteLater();
```

Comment:
small refactoring to controller() getters. Logic didn't change, but because our CameraController class is used more often than Qt's less dynamic_cast's will happen

Reply:
Is there a reason why we do not make (our) `CameraController` a sub class of `QAbstractCameraController`?

Reply:
I like your question. Before I intended to not repeat functionality of Qt in our code. But now since I already duplicated almost all properties to expose it to python, we can remove mixin and make it as you suggest.

Reply:
Either way works for me.  Qt prefers inheritance, but HPC code usually uses templates and/or mixins.  There is probably not a definite answer for everything.  But for the Qt code here, inheritance might not be wrong.